src/sources/generator.rs

```diff
@@ -50,7 +145,10 @@ impl SourceConfig for GeneratorConfig {
         shutdown: ShutdownSignal,
         out: Pipeline,
     ) -> crate::Result<super::Source> {
-        Ok(self.clone().generator(shutdown, out))
+        self.format
+            .validate()
+            .map_err(Into::into)
+            .map(|()| self.clone().generator(shutdown, out))
```

Comment:
I just realized, my previous comment was pretty silly, since the `?` operator does `into`. This is even simpler:
```suggestion
        self.format.validate()?;
        Ok(self.clone().generator(shutdown, out))
```